### PR TITLE
Efficient scatterplotting with ggplot2

### PR DESCRIPTION
**Pull request by votti**
This is the example script how to efficiently plot visually accurate scatterplots in R.

It simulates it's own testdata.

### CODE REVIEW

**Comment by JanaFischer** [comment]
Looks great to me! Already tried it on UMAPs and tSNEs of my own data and it works nicely and really fast.

File: snippets/varia/heatscatter_example/heatscatter_example.Rmd [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
+    set.seed(123)
+    d = lapply(specs, function(s){a_gaussian(s[1], s[2], s[3], num)})
+    d = rbindlist(d, idcol='g')
+    d[, g:= as.factor(g)]
+    d
+}
+
+```
+
+This simulates 4 partially overlapping gaussians and samples 50'000 points each, so 250'000 total:
```
Comment (nilseling) [inline snippets/varia/heatscatter_example/heatscatter_example.Rmd]: It simulates 5 Gaussians, right?

File: snippets/varia/heatscatter_example/heatscatter_example.Rmd [hunk 2]
```diff
@@ -0,0 +1,641 @@
...
+    scale_color_viridis(trans = "log10", option='inferno')+
+    ggtitle('Bin2D Points\nFake Points\nRuntime: ca 0.7 s')
+
+system.time({print(p)})
+
+```
+
+```{r}
+
+```
```
Comment (nilseling) [inline snippets/varia/heatscatter_example/heatscatter_example.Rmd:195]: Delete?

File: snippets/varia/heatscatter_example/heatscatter_example.Rmd [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+    stat_bin2d(aes(color=..count..), bins=100, geom='point', size=1.4, fill=1)+
+    scale_color_viridis(trans = "log10", option='inferno')+
+    ggtitle('Bin2D Points\nCoarser Fake Points\nRuntime: ca 0.5 s')
+
+system.time({print(p)})
+
+
+```
+
+Whenever bin2d is uese, also geom_hexbin can be used:
```
Comment (nilseling) [inline snippets/varia/heatscatter_example/heatscatter_example.Rmd]: used

**Review by nilseling — APPROVED** [review]
Really nice! I didn't fully get some things without trying them out myself but definitely a good way of visualising this! Just a couple of formatting comments.

File: snippets/varia/heatscatter_example/heatscatter_example.Rmd (hunk 1 above)
Comment (votti) [reply snippets/varia/heatscatter_example/heatscatter_example.Rmd]: Ah yeah, thanks will fix.

File: snippets/varia/heatscatter_example/heatscatter_example.Rmd (hunk 2 above)
Comment (votti) [reply snippets/varia/heatscatter_example/heatscatter_example.Rmd:195]: Yes!

File: snippets/varia/heatscatter_example/heatscatter_example.Rmd (hunk 3 above)
Comment (votti) [reply snippets/varia/heatscatter_example/heatscatter_example.Rmd]: Spelling things correctly is really not my forte - thanks will fix :)

**Review by ndamond — APPROVED** [review]
Great explanations, works nicely with my data.

**Pull request merged**